homeassistant/components/kef/media_player.py

```diff
@@ -0,0 +1,270 @@
+        maximum_volume,
+        volume_step,
+        standby_time,
+        inverse_speaker_mode,
+        sources,
+        ioloop,
+    ):
+        """Initialize the media player."""
+        self._name = name
+        self._sources = sources
+        self._speaker = AsyncKefSpeaker(
+            host,
+            port,
+            volume_step,
+            maximum_volume,
+            standby_time,
+            inverse_speaker_mode,
+            ioloop=ioloop,
+        )
+
+        self._state = STATE_UNKNOWN
+        self._muted = None
+        self._source = None
+        self._volume = None
+        self._is_online = None
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    async def async_update(self):
+        """Update latest state."""
+        _LOGGER.debug("Running async_update")
+        try:
+            self._is_online = await self._speaker.is_online()
+            if self._is_online:
+                (
+                    self._volume,
+                    self._muted,
+                ) = await self._speaker.get_volume_and_is_muted()
+                state = await self._speaker.get_state()
+                self._source = state.source
+                self._state = STATE_ON if state.is_on else STATE_OFF
+            else:
+                self._muted = None
+                self._source = None
+                self._volume = None
+                self._state = STATE_OFF
+        except (ConnectionRefusedError, ConnectionError, TimeoutError) as err:
+            _LOGGER.debug("Error in `update`: %s", err)
+            self._state = STATE_UNKNOWN
+
+    @property
+    def volume_level(self):
+        """Volume level of the media player (0..1)."""
+        return self._volume
+
+    @property
+    def is_volume_muted(self):
+        """Boolean if volume is currently muted."""
+        return self._muted
+
+    @property
+    def supported_features(self):
+        """Flag media player features that are supported."""
+        return SUPPORT_KEF
+
+    @property
+    def source(self):
+        """Name of the current input source."""
+        return self._source
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return self._sources
+
+    @property
+    def available(self):
+        """Return if the speaker is reachable online."""
+        return self._is_online
+
+    @property
+    def unique_id(self):
+        """Return the device unique id."""
+        return f"{self._speaker.host}:{self._speaker.port}"
+
+    @property
+    def icon(self):
+        """Return the device's icon."""
+        return "mdi:speaker-wireless"
+
+    @property
+    def should_poll(self):
+        """It's possible that the speaker is controlled manually."""
+        return True
+
+    @property
+    def force_update(self):
```

Comment:
Please remove this.

Reply:
Fixed in 38c25cc25ec257caf2098404040638e20a290160.